packages/venia-ui/lib/components/AuthBar/authBar.js

```diff
@@ -1,25 +1,31 @@
-import React, { useCallback } from 'react';
+import React from 'react';
 import { bool, func, shape, string } from 'prop-types';
-import { useUserContext } from '@magento/peregrine/lib/context/user';
 
 import { mergeClasses } from '../../classify';
 import Button from '../Button';
 import UserChip from './userChip';
 import defaultClasses from './authBar.css';
+import { useAuthBar } from '@magento/peregrine/lib/mixins/AuthBar/useAuthBar';
 
 const AuthBar = props => {
-    const { disabled, showMyAccount, showSignIn } = props;
-    const [{ currentUser, isSignedIn }] = useUserContext();
     const classes = mergeClasses(defaultClasses, props.classes);
 
-    const handleClick = useCallback(() => {
-        showSignIn();
-    }, [showSignIn]);
+    const {
+        currentUser,
+        handleShowMyAccount,
+        handleSignIn,
+        isSignedIn,
+        isSignInDisabled
+    } = useAuthBar(props);
 
     const child = isSignedIn ? (
-        <UserChip user={currentUser} showMyAccount={showMyAccount} />
+        <UserChip user={currentUser} showMyAccount={handleShowMyAccount} />
     ) : (
-        <Button disabled={!!disabled} priority="high" onClick={handleClick}>
```

Comment:
Looks like we lost the `!!` for this button being disabled.  Will the `disabled` prop ever not be a boolean?

Reply:
Looks like it will always be a boolean 👍 

[Navigation](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/Navigation/navigation.js) is the only place that calls `AuthBar`:

```js
const hasModal = view !== 'MENU';

// ...

<AuthBar
    disabled={hasModal}
    showMyAccount={showMyAccount}
    showSignIn={showSignIn}
/>
```